js/repl/Repl.js

```diff
@@ -447,6 +467,12 @@ class Repl extends React.Component {
     // This prevents frequent updates while a user is typing.
     this._compileToState(code);
   };
+
+  _updateStatus(status: Object) {
+    this.setState({
+      status: Object.assign({}, this.state.status, status),
```

Comment:
This should use the [updater-function format of `setState`](https://reactjs.org/docs/react-component.html#setstate) to avoid potential stale values in `this.state.status`.

Reply:
Actually, based on my comment on `js/repl/Utils.js` (below) this logic might be able to go away entirely.